include/fixed_containers/fixed_red_black_tree_view.hpp

```diff
@@ -0,0 +1,386 @@
+        using reference = value_type&;
+        using const_reference = const value_type&;
+        using iterator_category = std::forward_iterator_tag;
+
+        constexpr Iterator(const std::byte* ptr,
+                           std::size_t elem_size_bytes,
+                           std::size_t max_size_bytes,
+                           Compactness compactness,
+                           StorageType storage_type,
+                           bool end = false) noexcept
+          : base_(ptr)
+          , elem_size_bytes_(elem_size_bytes)
+          , max_size_bytes_(max_size_bytes)
+          , compactness_(compactness)
+          , storage_type_(storage_type)
+          , storage_elem_size_bytes_(storage_elem_size_bytes())
+          , index_(end ? NULL_INDEX : min_index())
+          , cur_pointer_(node_pointer(index_))
+        {}
+
+        constexpr Iterator() noexcept
+          : Iterator(nullptr, {}, {}, {}, {}, false)
+        {}
+
+        constexpr Iterator(const Iterator&) noexcept = default;
+        constexpr Iterator(Iterator&&) noexcept = default;
+        constexpr Iterator& operator=(const Iterator &) noexcept = default;
+        constexpr Iterator& operator=(Iterator&&) noexcept = default;
+
+        constexpr const_reference operator*() const
+        {
+            return cur_pointer_;
+        }
+        
+        constexpr const_pointer operator->()
+        {
+            return &cur_pointer_;
+        }
+        
+        constexpr Iterator& operator++()
+        {
+            index_ = successor(index_);
+            cur_pointer_ = node_pointer(index_);
+            return *this;
+        }
+
+        constexpr Iterator operator++(int) & noexcept
+        {
+            auto tmp = *this;
+            index_ = successor(index_);
+            cur_pointer_ = node_pointer(index_);
+            return tmp;
+        }
+        
+        constexpr bool operator==(const Iterator& other) const
+        {
+            return (
+                base_ == other.base_ &&
+                index_ == other.index_
+            );
+        }
+        
+    private:
+        /**
+         * Calculate the pointer to a tree node at the provided storage index.
+         */
+        constexpr const std::byte* node_pointer(std::size_t i) const
+        {
+            switch (storage_type_)
+            {
+            case StorageType::FIXED_INDEX_POOL:
+                return (iov_array_base() + i * storage_elem_size_bytes_);
+
+            case StorageType::FIXED_INDEX_CONTIGUOUS:
+                return (contiguous_array_base() + i * storage_elem_size_bytes_);
+            }
+        }
+
+        /**
+         * Traverse the tree to find the index corresponding to the minimum node.
+         */
+        NodeIndex min_index() const
+        {
+            auto i = root_index();
+            if (i == NULL_INDEX)
+            {
+                return NULL_INDEX;
+            }
+
+            NodeIndex left;
```

Comment:
Nit: uninitialized variable.

Reply:
Intentional. I initialize it inside the `if` clause. However, I can make this look neater if I just declare the variable inside the `if` clause as well, so I will do that.

Reply:
2821f91